x-pack/plugins/security_solution/public/management/cypress/e2e/artifacts.cy.ts

```diff
@@ -0,0 +1,181 @@
+  EXCEPTION_LIST_URL,
+} from '@kbn/securitysolution-list-constants';
+import { BASE_ENDPOINT_ROUTE } from '../../../../common/endpoint/constants';
+import { login, loginWithRole, ROLE } from '../tasks/login';
+
+import { type FormAction, getArtifactsListTestsData } from './mocks/artifacts_page';
+import { runEndpointLoaderScript } from '../tasks/run_endpoint_loader';
+
+const removeAllArtifacts = () => {
+  for (const listId of ENDPOINT_ARTIFACT_LIST_IDS) {
+    cy.request({
+      method: 'DELETE',
+      url: `${EXCEPTION_LIST_URL}?list_id=${listId}&namespace_type=agnostic`,
+      headers: { 'kbn-xsrf': 'kibana' },
+      failOnStatusCode: false,
+    });
+  }
+};
+
+const loginWithWriteAccess = (url: string) => {
+  loginWithRole(ROLE.analyst_hunter);
+  cy.visit(url);
+};
+
+const loginWithReadAccess = (url: string) => {
+  loginWithRole(ROLE.t2_analyst);
+  cy.visit(url);
+};
+
+const loginWithoutAccess = (url: string) => {
+  loginWithRole(ROLE.t1_analyst);
+  cy.visit(url);
+};
+
+// Checks for Endpoint data and creates it if needed
+const loadEndpointDataForEventFiltersIfNeeded = () => {
+  cy.request({
+    method: 'POST',
+    url: `${BASE_ENDPOINT_ROUTE}/suggestions/eventFilters`,
+    body: {
+      field: 'agent.type',
+      query: '',
+    },
+    headers: { 'kbn-xsrf': 'kibana' },
+    failOnStatusCode: false,
+  }).then(({ body }) => {
+    if (isEmpty(body)) {
+      runEndpointLoaderScript();
+    }
+  });
+};
+
+const runAction = (action: FormAction) => {
+  if (action.type === 'click') {
+    cy.getBySel(action.selector).click();
+  } else if (action.type === 'input') {
+    cy.getBySel(action.selector).type(action.value || '');
+  } else if (action.type === 'clear') {
+    cy.getBySel(action.selector).clear();
+  } else if (action.type === 'customClick') {
+    cy.get(action.selector).click();
+  } else if (action.type === 'removeItem') {
+    cy.get(`[data-test-subj="${action.selector}"] > span > button`).click();
+  }
+};
+
+describe('Artifacts pages', () => {
+  before(() => {
+    login();
+    loadEndpointDataForEventFiltersIfNeeded();
+    // Clean artifacts data
+    removeAllArtifacts();
+  });
+
+  after(() => {
+    // Clean artifacts data
+    removeAllArtifacts();
+  });
+
+  for (const testData of getArtifactsListTestsData()) {
+    describe(`When on the ${testData.title} entries list`, () => {
+      it(`no access - should not show empty state page and the add button does not exists`, () => {
+        loginWithoutAccess(`/app/security/administration/${testData.urlPath}`);
+        cy.getBySel('noPrivilegesPage').should('exist');
+        cy.getBySel('empty-page-feature-action').should('exist');
+        cy.getBySel(testData.emptyState).should('not.exist');
+        cy.getBySel(`${testData.pagePrefix}-emptyState-addButton`).should('not.exist');
+      });
+
+      it(`read - should show empty state page if there is no ${testData.title} entry and the add button does not exists`, () => {
```

Comment:
nit: `...does not exist`

Reply:
https://github.com/elastic/kibana/pull/148404/commits/9a09d04d6edc1ca8995268637a731217ad4f1907